src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizerOfT.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.Framework.Localization;
+
+namespace Microsoft.AspNet.Mvc.Localization
+{
+    /// <summary>
+    /// Represents an <see cref="IHtmlLocalizer"/> that provides strings for <see cref="TResourceSource"/>.
+    /// </summary>
+    /// <typeparam name="TResourceSource">The <see cref="System.Type"/> to provide strings for.</typeparam>
+    public class HtmlLocalizer<TResourceSource> : IHtmlLocalizer<TResourceSource>
+    {
+        private readonly IHtmlLocalizer _localizer;
+
+        public HtmlLocalizer(IHtmlLocalizerFactory factory)
+        {
+            _localizer = factory.Create(typeof(TResourceSource));
+        }
+
+        /// <inheritdoc />
+        public virtual IHtmlLocalizer WithCulture(CultureInfo culture) => _localizer.WithCulture(culture);
```

Comment:
Need a strong does of `[NotNull]` in this class and / or in the `IHtmlLocalizer` and `IHtmlLocalizer<TResourceSource>` interfaces.